compiler/src/AST/Source.hs

```diff
@@ -124,14 +125,16 @@ data Type_
 
 -- MODULE
 
+type SourceOrder = Int
+
 data Module = Module
   { _name :: Maybe (A.Located Name),
     _exports :: A.Located Exposing,
     _docs :: Docs,
     _imports :: [Import],
-    _values :: [A.Located Value],
-    _unions :: [A.Located Union],
-    _aliases :: [A.Located Alias],
+    _values :: [(SourceOrder, A.Located Value)],
+    _unions :: [(SourceOrder, A.Located Union)],
+    _aliases :: [(SourceOrder, A.Located Alias)],
```

Comment:
I think the `fmap snd`'s you're talking about are fmap'ing a List, and have type `[(sourceOrder, A.Located a)] -> A.Located a`, so having a `A.SourceOrderWithLocation a` type would still need an `fmap someNewFunction` with `someNewFunction : A.SourceOrderWithLocation a -> A.Located a`, or maybe in some places where `A.toValue` is used later on, we'd have `someOtherNewFunction : A.SourceOrderWithLocation a -> a`.

So I think the result would just change `fmap snd` into `fmap someNewFunction`, which I think does have different trade-offs, but doesn't seem strictly better to me.  But if you think having some more named functions would be worth it to avoid using more tuples, then I can do that.  What do you think?

Reply:
Just keep it as is.

A part of me is thinking "there's got to be a way to do this without adding fmaps all over", but I'm sure you're right in your assessment that it would include different trade offs.

The code is easy enough to understand, so just keep it.

Reply:
maybe the `fmap`'s are a smell that the `SourceOrder`'s are being passed down further than they need to be?  I'll keep an eye out for that when I fix the other feedback.